pilot/pkg/model/service.go

```diff
@@ -477,6 +490,23 @@ func ParseServiceKey(s string) (hostname string, ports PortList, labels LabelsCo
 	return
 }
 
+// BuildSubsetKey generates a unique string referencing service instances for a given service name, a subset and a port.
```

Comment:
Nit, I see that it's actually building the cluster name. Consider renaming it accordingly.